src/loaders/gamefile/mod.rs

```diff
@@ -193,13 +196,40 @@ impl GameArchive {
     }
 }
 
-#[derive(Default)]
 pub struct GameFileLoader {
     archives: Vec<GameArchive>,
+    game_file_settings: GameFileSettings,
     lua_files: Vec<String>,
 }
 
+impl Default for GameFileLoader {
+    fn default() -> Self {
+        let game_file_settings = GameFileSettings::new();
+        let archives: Vec<GameArchive> = Vec::new();
+        let lua_files: Vec<String> = Vec::new();
+
+        Self {
+            archives,
+            game_file_settings,
+            lua_files,
+        }
+    }
+}
+
 impl GameFileLoader {
+    pub fn add_archives_from_settings(&mut self) {
+        #[cfg(feature = "debug")]
+        let timer = Timer::new_dynamic(format!("load game files list from settings"));
+
+        for path in &self.game_file_settings.archives {
+            let game_archive = GameArchive::load(&path, &mut self.lua_files);
+            self.archives.insert(0, game_archive);
+        }
```

Comment:
I'm not so happy with this implementation, do you have suggestions? Borrowing the mutable to use `Self.add_archive(self, path)` doesn't work and, I dunno how to implement it in a better way.

Reply:
I also ran into this issue when looking into your PR. I ended up giving the `add_archive` function the actual `GameArchive` object instead of a borrowed `&str` to fix it